### PR TITLE
WRO-12243: [Codecov] Increase coverage for VideoPlayer

### PR DESCRIPTION
**Pull request by sjro**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Increased test coverage of VideoPlayer

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Added unit test for VideoPlayer, MediaSlider and util.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-12243

### Comments
Enact-DCO-1.0-Signed-off-by: SJ RO (sj.ro@lge.com)

### CODE REVIEW

File: MediaPlayer/tests/MediaSlider-specs.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		render(
+			<MediaSlider data-testid="mediaslider-id" onFocus={handleFocus} />
+		);
+
+		const slider = screen.getByTestId('mediaslider-id');
+		focus(slider);
+
+		expect(handleFocus).toBeCalled();
+	});
+	test('should fire `onBlur` on blur', () => {
```
Comment (MikyungKim) [inline MediaPlayer/tests/MediaSlider-specs.js:27]: Please add an empty line between each test.

File: MediaPlayer/tests/MediaSlider-specs.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+const blur = (slider) => fireEvent.blur(slider);
+const focus = (slider) => fireEvent.focus(slider);
+const activate = (slider) => fireEvent.keyUp(slider, {keyCode: 13});
+const keyDown = (keyCode) => (slider) => fireEvent.keyDown(slider, {keyCode});
+
+const leftKeyDown = keyDown(37);
+const rightKeyDown = keyDown(39);
+
+describe('MediaSlider', () => {
+	test('should fire `onFocus` when focused', () => {
```
Comment (MikyungKim) [inline MediaPlayer/tests/MediaSlider-specs.js:15]:
```suggestion
	test('should forward `onFocus` when focused', () => {
```

File: MediaPlayer/tests/MediaSlider-specs.js (hunk 1 above)
Comment (MikyungKim) [inline MediaPlayer/tests/MediaSlider-specs.js:27]:
```suggestion
	test('should forward `onBlur` on blur', () => {
```

File: MediaPlayer/tests/MediaSlider-specs.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+		render(
+			<MediaSlider data-testid="mediaslider-id" onBlur={handleBlur} />
+		);
+
+		const slider = screen.getByTestId('mediaslider-id');
+		blur(slider);
+
+		expect(handleBlur).toBeCalled();
+	});
+	test('should fire `onKeyDown` when key pressed', () => {
```
Comment (MikyungKim) [inline MediaPlayer/tests/MediaSlider-specs.js:39]:
```suggestion
	test('should forward `onKeyDown` when key pressed', () => {
```

File: MediaPlayer/tests/MediaSlider-specs.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+		const slider = screen.getByTestId('mediaslider-id');
+
+		rightKeyDown(slider);
+		leftKeyDown(slider);
+
+		const expected = 2;
+
+		expect(handleKeyDown).toBeCalledTimes(expected);
+	});
+	test('should fire `onKeyUp` when active', () => {
```
Comment (MikyungKim) [inline MediaPlayer/tests/MediaSlider-specs.js:55]:
```suggestion
	test('should forward `onKeyUp` when active', () => {
```

File: MediaPlayer/tests/MediaSlider-specs.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+		);
+
+		const slider = screen.getByTestId('mediaslider-id');
+
+		activate(slider);
+
+		expect(handleKeyUp).toBeCalled();
+	});
+
+});
```
Comment (MikyungKim) [inline MediaPlayer/tests/MediaSlider-specs.js:69]: Could we add testing mouse events also?
I confirmed that if we use something like `fireEvent.mouseOver(slider, {clientX: 10});`, and check `onKnobMove` event is fired, we could increase the coverage more.
I suggest considering firing mouseOver, mouseMove, and mouseOut.

File: MediaPlayer/tests/util-specs.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+		});
+
+		test('should return hour unit string as 00 when seconds is less than 3600 sec and includeHour is true', () => {
+			const seconds = 100;
+			const expected = '00:01:40';
+			const actual = secondsToTime(seconds, getDurFmt('en'), {includeHour: true});
+
+			expect(actual).toEqual(expected);
+		});
+		test('should not return hour unit string when seconds is less than 3600 sec and includeHour is false', () => {
```
Comment (MikyungKim) [inline MediaPlayer/tests/util-specs.js:50]: Please add an empty line between each test.

File: MediaPlayer/tests/util-specs.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+			const seconds = 100;
+			const expected = '01:40';
+			const actual = secondsToTime(seconds, getDurFmt('en'), {includeHour: false});
+
+			expect(actual).toEqual(expected);
+		});
+		test('should return 00:00:00 when instance is null and includeHour is true', () => {
```
Comment (MikyungKim) [inline MediaPlayer/tests/util-specs.js:57]:
```suggestion
		test('should return 00:00:00 when instance of a `ilib.DurationFmt` object is null and includeHour is true', () => {
```

File: MediaPlayer/tests/util-specs.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+			const seconds = 100;
+			const expected = '00:00:00';
+			const actual = secondsToTime(seconds, null, {includeHour: true});
+
+			expect(actual).toEqual(expected);
+		});
+		test('should return 00:00 when instance is null and includeHour is false', () => {
```
Comment (MikyungKim) [inline MediaPlayer/tests/util-specs.js:64]:
```suggestion
		test('should return 00:00 when instance of a `ilib.DurationFmt` object is null and includeHour is false', () => {
```

File: MediaPlayer/tests/util-specs.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
+import {memoize} from '@enact/core/util';
+import DurationFmt from 'ilib/lib/DurationFmt';
+
+import {parseTime, secondsToTime} from '../util';
```
Comment (MikyungKim) [inline MediaPlayer/tests/util-specs.js:4]: Could we test `secondsToPeriod` and `countReactChildren` here too? Even though it seems been covered by other tests.

File: VideoPlayer/tests/VideoPlayer-specs.js [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+
+		const playingEvent = createEvent('playing', video);
+		fireEvent(video, playingEvent);
+
+		const expected = {type: 'playing'};
+		const actual = handlePlaying.mock.calls.length && handlePlaying.mock.calls[0][0];
+
+		expect(actual).toMatchObject(expected);
+	});
+	test('should fire `onControlsAvailable` with `onControlsAvailable` type when screen clicked', () => {
```
Comment (MikyungKim) [inline VideoPlayer/tests/VideoPlayer-specs.js:29]: Please add an empty line between each test.

File: MediaPlayer/tests/MediaSlider-specs.js (hunk 1 above)
Comment (sjro) [reply MediaPlayer/tests/MediaSlider-specs.js]: added it

File: MediaPlayer/tests/MediaSlider-specs.js (hunk 2 above)
Comment (sjro) [reply MediaPlayer/tests/MediaSlider-specs.js]: Fixed it

File: MediaPlayer/tests/MediaSlider-specs.js (hunk 1 above)
Comment (sjro) [reply MediaPlayer/tests/MediaSlider-specs.js]: Fixed it.

File: MediaPlayer/tests/MediaSlider-specs.js (hunk 3 above)
Comment (sjro) [reply MediaPlayer/tests/MediaSlider-specs.js]: Fixed it

File: MediaPlayer/tests/MediaSlider-specs.js (hunk 4 above)
Comment (sjro) [reply MediaPlayer/tests/MediaSlider-specs.js]: Fixed it.

File: MediaPlayer/tests/MediaSlider-specs.js (hunk 5 above)
Comment (sjro) [reply MediaPlayer/tests/MediaSlider-specs.js:113]: I added mouseOver and mouseMove.
But there is no expected case when mouseOut event occur.

File: MediaPlayer/tests/util-specs.js (hunk 6 above)
Comment (sjro) [reply MediaPlayer/tests/util-specs.js:84]: Added it

File: MediaPlayer/tests/util-specs.js (hunk 7 above)
Comment (sjro) [reply MediaPlayer/tests/util-specs.js]: Fixed it.

File: MediaPlayer/tests/util-specs.js (hunk 8 above)
Comment (sjro) [reply MediaPlayer/tests/util-specs.js]: Fixed it.

File: MediaPlayer/tests/util-specs.js (hunk 9 above)
Comment (sjro) [reply MediaPlayer/tests/util-specs.js]: Added the case.

File: VideoPlayer/tests/VideoPlayer-specs.js [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+		const handleControlsAvailable = jest.fn();
+
+		render(
+			<VideoPlayer data-testid="videoplayer-id" onControlsAvailable={handleControlsAvailable} />
+		);
+
+		const overlay = screen.getByTestId('videoplayer-id').nextElementSibling;
+		userEvent.click(overlay);
+
+		const slider = screen.getByRole('slider', {hidden: true});  // add to increase code coverage
```
[the remark this answers is not in the document]
Comment (sjro) [reply VideoPlayer/tests/VideoPlayer-specs.js]: The comment moved the next line.
focus(slider) is affected handleSliderFocus() lines.

File: VideoPlayer/tests/VideoPlayer-specs.js (hunk 10 above)
Comment (sjro) [reply VideoPlayer/tests/VideoPlayer-specs.js:30]: Added it

File: MediaPlayer/tests/MediaSlider-specs.js [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+			<MediaSlider data-testid="mediaslider-id" onFocus={handleFocus} />
+		);
+
+		const slider = screen.getByTestId('mediaslider-id');
+		focus(slider);
+
+		expect(handleFocus).toBeCalled();
+	});
+
+	test('should forward `onBlur` on blur', () => {
```
Comment (MikyungKim) [inline MediaPlayer/tests/MediaSlider-specs.js:41]:
```suggestion
	test('should forward `onBlur` when blurred', () => {
```

File: MediaPlayer/tests/util-specs.js [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+
+const getDurFmt = (locale) => {
+	if (typeof window === 'undefined') return null;
+
+	return memoGetDurFmt(locale);
+};
+
+describe('util', () => {
+	describe('countReactChildren', () => {
+		test('should return the chilren node number', () => {
```
Comment (MikyungKim) [inline MediaPlayer/tests/util-specs.js:19]:
```suggestion
		test('should return the number of children node', () => {
```

File: MediaPlayer/tests/MediaSlider-specs.js [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+
+		render(
+			<MediaSlider data-testid="mediaslider-id" onKnobMove={handleKnobMove} />
+		);
+
+		const slider = screen.getByTestId('mediaslider-id');
+
+		focus(slider);
+
+		const touches = [{clientX:10}];
```
Comment (MikyungKim) [inline MediaPlayer/tests/MediaSlider-specs.js:125]:
```suggestion
		const touches = [{clientX: 10}];
```

File: MediaPlayer/tests/MediaSlider-specs.js (hunk 12 above)
Comment (sjro) [reply MediaPlayer/tests/MediaSlider-specs.js]: fixed it

File: MediaPlayer/tests/util-specs.js (hunk 13 above)
Comment (sjro) [reply MediaPlayer/tests/util-specs.js]: fixed it.

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**